src/cascade/input_data/db/configuration.py

```diff
@@ -2,30 +2,111 @@
 import logging
 
 from cascade.core.db import cursor
+from cascade.input_data.configuration import SettingsError
+from cascade.input_data.configuration.form import Configuration
 from cascade.testing_utilities import make_execution_context
 from cascade.core.db import latest_model_version
 
 CODELOG = logging.getLogger(__name__)
+MATHLOG = logging.getLogger(__name__)
+
+
+def load_settings(meid=None, mvid=None, settings_file=None):
+    CODELOG.debug(f"meid {meid} mvid {mvid} settings {settings_file}")
+    if len([c for c in [meid, mvid, settings_file] if c is not None]) != 1:
+        raise ValueError(
+            "Must supply exactly one of mvid, meid or settings_file")
+    if meid:
+        raw_settings, found_mvid = load_raw_settings_meid(meid)
+    elif mvid:
+        raw_settings, found_mvid = load_raw_settings_mvid(mvid)
+    elif settings_file:
+        raw_settings, found_mvid = load_raw_settings_file(settings_file)
+    else:
+        raise RuntimeError(f"Either meid, mvid, or file must be specified.")
 
+    return json_settings_to_frozen_settings(raw_settings, found_mvid)
 
-def settings_for_model(meid=None, mvid=None):
-    if len([c for c in [meid, mvid] if c is not None]) != 1:
-        raise ValueError("Must supply either mvid or meid")
 
-    if mvid is not None:
-        mvid = mvid
-    else:
-        modelable_entity_id = meid
-        ec = make_execution_context(modelable_entity_id=modelable_entity_id)
-        mvid = latest_model_version(ec)
-        CODELOG.info(f"No model version specified so using the latest version for model {meid} which is {mvid}")
+def load_raw_settings_meid(modelable_entity_id):
+    """
+    Given a meid, get settings for the latest corresponding mvid.
+
+    Args:
+        modelable_entity_id (int,str): The MEID.
+    Returns:
+        dict: Settings as a JSON dictionary of dictionaries.
+        int: Model version ID that is the latest one associated with this meid.
+    """
+    ec = make_execution_context(modelable_entity_id=modelable_entity_id)
```

Comment:
This is dangerous because it looses the prod/dev distinction that may be in place in the outer code. The context should be passed in. This is unlikely to cause problems now but it's brittle and bug prone.

Reply:
I know that you are just copying the pattern I established, but I was lazy and wrong.

Reply:
You need the execution context to get the settings, but you need the settings to make the whole execution context. It looks like it would help to have one execution context that is the environment of the application. That gets passed into here. Then have a different one, or a different part of it, that is says where we are in the cascade hierarchy. That's what gets made. I don't want to go after that in this PR. That's a reorganization.

Reply:
NVMD. I made the execution context and then appended to it as I went.